cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java

```diff
@@ -142,4 +165,132 @@ public void setMissingIncludeWarningsEnabled(boolean enabled){
+      } else {
+        LOG.error("Compilation log not found: '{}'", buildLog.getAbsolutePath());
+      }    
+    }
+  }
+
+  private void parseVCppLog(File buildLog, String charsetName) {
+
+      try {
+        BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(buildLog), charsetName));
+        String line;
+        String currentProjectPath = "";
+        while ((line = br.readLine()) != null) {
+          if (line.startsWith("  INCLUDE=")) { // handle environment includes 
+            String[] includes = line.split("=")[1].split(";");
+            for(String include : includes) {
+              if (!uniqueIncludes.contains(include)) {
+                uniqueIncludes.add(include);
+              }
+            }
+          }
+          
+          // get base path of project to make 
+          // Target "ClCompile" in file "C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.CppCommon.targets" from project "D:\Development\SonarQube\cxx\sonar-cxx\integration-tests\testdata\googletest_bullseye_vs_project\PathHandling.Test\PathHandling.Test.vcxproj" (target "_ClCompile" depends on it):
+          if (line.startsWith("Target \"ClCompile\" in file")) {
+            currentProjectPath = line.split("\" from project \"")[1].split("\\s+")[0].replace("\"", "");              
+          }
+          
+          if (line.contains("C:\\Program Files (x86)\\BullseyeCoverage\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);
+          }
+          
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 10.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 11.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 11.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);       
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 12.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 12.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);        
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 14.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);        
+          }  
+        }
+        br.close();
+      } catch (IOException ex) {
+        LOG.error("Cannot parse build log", ex);
+      }
+  }
+
+  private void parseVCppCompilerCLLine(String line, String projectPath) {
```

Comment:
Yes this is correct, the next step is to implement settings per file based
approach. But will not have time until the end of the month to get this .
so we can leave it for next major release. Later once this is on, we can
support #421 more easily

On Thu, Apr 9, 2015, 21:46 Günter Wirth <notifications@github.com> wrote:

> In cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java
> <https://github.com/wenns/sonar-cxx/pull/399#discussion_r28090534>:
>
> > +          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 12.0\\VC\\bin\\CL.exe") ||
> > +                  line.contains("C:\\Program Files\\Microsoft Visual Studio 12.0\\VC\\bin\\CL.exe")) {
> > +            parseVCppCompilerCLLine(line, currentProjectPath);
> > +          }
> > +          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\CL.exe") ||
> > +                  line.contains("C:\\Program Files\\Microsoft Visual Studio 14.0\\VC\\bin\\CL.exe")) {
> > +            parseVCppCompilerCLLine(line, currentProjectPath);
> > +          }
> > +        }
> > +        br.close();
> > +      } catch (IOException ex) {
> > +        LOG.error("Cannot parse build log", ex);
> > +      }
> > +  }
> > +
> > +  private void parseVCppCompilerCLLine(String line, String projectPath) {
>
> Not sure if this approach is really working. In VS each source file could
> have own settings. My understanding of this function is that you are
> collecting the settings of all files which could be wrong.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wenns/sonar-cxx/pull/399/files#r28090534>.
>